R/taxonomy.R

```diff
@@ -197,6 +209,68 @@ setMethod("checkTaxonomy", signature = c(x = "SummarizedExperiment"),
     }
 )
 
+#' @rdname taxonomy-methods
+#' @importFrom utils assignInMyNamespace
+#' @aliases checkTaxonomy
+#' @export
+# Function to set taxonomy ranks
+setTaxonomyRanks <- function(ranks) {
+    ranks <- tolower(ranks)
+    # Check if rank is a character vector with length >= 1
+    if (!is.character(ranks) || length(ranks) < 1 
+        || any(ranks == "" | ranks == " " | ranks == "\t" | ranks == "-" | ranks == "_")
+        || any(grepl("\\s{2,}", ranks))) {
+        stop("Input 'rank' should be a character vector with non-empty strings,
+             no spaces, tabs, hyphens, underscores, and non-continuous spaces."
+             , call. = FALSE)
+    }
+    
+    assignInMyNamespace("TAXONOMY_RANKS", ranks)
+}
+
+
+#' @rdname taxonomy-methods
+#'@export
+# Function to get taxonomy ranks
+getTaxonomyRanks <- function() {
+    return(TAXONOMY_RANKS)
+}
+
+.set_taxonomy_ranks <- function(tse, ranks, set.ranks = FALSE) {
```

Comment:
So I believe this function is intended to use as internal function (Added to end of `loadFromMetaphlan` for instance)?

Add `ranks = NULL`, and `...`